### DISCUSSION TITLE
How to have more than one layout

### DISCUSSION THREAD

**Discussion by ahakem** [vercel/next.js#13251] [Help] [answered]
I have a site I'm developing have some screens without header and footer.
the only way in my mind is to load the header and footer if the routers`!==` the targeted pages

but looking for a better solution as maybe 40% of the site will be this case

**Comment by helloworld** [comment, marked as answer]
Here's an in-depth tutorial on different approaches you can use and the tradeoffs between them: https://adamwathan.me/2019/10/17/persistent-layout-patterns-in-nextjs/

In short, the different approaches you can use are:

1. Higher-order layout components, so that you can export individual pages with a layout

```
export default withAppLayout(Home);
```

2. In `_app.js`, render a different layout based on route:

```
const router = useRouter()
if (router.pathname.startsWith('/sectionA/')) {
      return (
        <LayoutWithoutHeaderAndFooter>
            <Component {...pageProps}></Component>
        </LayoutWithoutHeaderAndFooter>
      )
}

return (
      <Layout>
        <Component {...pageProps}></Component>
      </Layout>
 )
```

3. Add a static `layout` property to your page, and access it in `_app.js` to determine which layout to use:

`myPage.js`:
```
const MyPage = () => (
  <div>Content</div>
)

MyPage.layout = LayoutWithoutHeaderAndFooter

export default MyPage
```

`_app.js`:
```
const { Component, pageProps } = this.props
const Layout = Component.layout || (children => <>{children}</>)

return (
      <Layout>
        <Component {...pageProps}></Component>
      </Layout>
)
```

**Reply by tayyabferozi** [reply]
In 3, under _app.js, it should be this:

`const Layout = Component.layout || (({ children }) => (<>{children}</>));`

instead of this:

`const Layout = Component.layout || (children => <>{children}</>)`

notice the curly braces
